DUI3-DX/Converters/Revit/Speckle.Converters.RevitShared/ToHost/Raw/Geometry/PolylineConverterToHost.cs

```diff
@@ -0,0 +1,96 @@
+using Speckle.Converters.RevitShared.Helpers;
+using Speckle.Converters.RevitShared.Services;
+using Speckle.Core.Logging;
+
+namespace Speckle.Converters.RevitShared.ToSpeckle;
+
+public class PolylineConverterToHost : ITypedConverter<SOG.Polyline, DB.CurveArray>
+{
+  private readonly ITypedConverter<SOG.Line, DB.Line> _lineConverter;
+  private readonly ScalingServiceToHost _scalingService;
+  private readonly IRevitConversionContextStack _contextStack;
+
+  public PolylineConverterToHost(
+    ITypedConverter<SOG.Line, DB.Line> lineConverter,
+    ScalingServiceToHost scalingService,
+    IRevitConversionContextStack contextStack
+  )
+  {
+    _lineConverter = lineConverter;
+    _scalingService = scalingService;
+    _contextStack = contextStack;
+  }
+
+  public CurveArray Convert(Polyline target)
+  {
+    var curveArray = new CurveArray();
+    if (target.value.Count == 6)
+    {
+      // Polyline is actually a single line
+      TryAppendLineSafely(curveArray, new SOG.Line(target.value, target.units));
+    }
+    else
+    {
+      var pts = target.GetPoints();
+      var lastPt = pts[0];
+      for (var i = 1; i < pts.Count; i++)
+      {
+        var success = TryAppendLineSafely(curveArray, new SOG.Line(lastPt, pts[i], target.units));
+        if (success)
+        {
+          lastPt = pts[i];
+        }
+      }
+
+      if (target.closed)
+      {
+        TryAppendLineSafely(curveArray, new SOG.Line(pts[^1], pts[0], target.units));
+      }
+    }
+    return curveArray;
+  }
+
+  /// <summary>
+  /// Checks if a Speckle <see cref="SOG.Line"/> is too sort to be created in Revit.
+  /// </summary>
+  /// <remarks>
+  /// The length of the line will be computed on the spot to ensure it is accurate.
+  /// </remarks>
+  /// <param name="line">The <see cref="SOG.Line"/> to be tested.</param>
+  /// <returns>true if the line is too short, false otherwise.</returns>
+  public bool IsLineTooShort(SOG.Line line)
+  {
+    var scaleToNative = _scalingService.ScaleToNative(SOG.Point.Distance(line.start, line.end), line.units);
+    return scaleToNative < _contextStack.Current.Document.Application.ShortCurveTolerance;
+  }
+
+  /// <summary>
+  /// Attempts to append a Speckle <see cref="SOG.Line"/> onto a Revit <see cref="CurveArray"/>.
+  /// This method ensures the line is long enough to be supported.
+  /// It will also convert the line to Revit before appending it to the <see cref="CurveArray"/>.
+  /// </summary>
+  /// <param name="curveArray">The revit <see cref="CurveArray"/> to add the line to.</param>
+  /// <param name="line">The <see cref="SOG.Line"/> to be added.</param>
+  /// <returns>True if the line was added, false otherwise.</returns>
+  public bool TryAppendLineSafely(CurveArray curveArray, SOG.Line line)
+  {
+    if (IsLineTooShort(line))
+    {
+      // poc : logging "Some lines in the CurveArray where ignored due to being smaller than the allowed curve length."
+      return false;
+    }
+    try
```

Comment:
can we have a whitespace here also, more importantly...
- Why the try catch?
- Why are we swallowing siently?

I know we're bring this stuff across but I think we should review some of these slightly bogus choices. If the convert() files, why are we continuing?

@JR-Morgan has strong feelings on this and @clairekuang I think you and @didimitrie need to establish the whole fail policy, as this is succeed but some things went bad and oh we didn't tell you. I don't think that's the right approach tbh...

Reply:
The long-term intention is that we wouldn't swallow silently, as indicated by the poc : logging comment. Have we decided on a method of logging that we are happy with while I was gone? If so, then I'll change these lines.

My assumption is that we added these guard clauses and try-catch statements for a reason, so I left them here. I could just remove them if we think that is a better idea.